### PR TITLE
Fixes #1001. Hide NSFW images by default.

### LINKED ISSUE #1001: [feature-request] Tag certain bugs as NSFW and hide images by default?
Often we get reports related to NSFW websites. They can sometimes contain screenshots that are adult-oriented. I wonder if we could add a `status-nsfw` (or similar, status is probably the wrong prefix), and it would overlay an opaque graphic over any embedded images. You could then click the image to display it, but they would be hidden by default.

(or maybe we have a settings panel somewhere where you opt-in to always see, dunno)

Useful if you're debugging websites and have children around, or are in a public space.

One example is https://github.com/webcompat/web-bugs/issues/2428.

Low priority, but it might be a nice feature to add.

### PR DESCRIPTION
**Pull request by miketaylr**
(I just realized, this only works for issues with a "nsfw" label already applied. Follow up bug #1086 was filed to make it work as soon as you apply the label)

r? @magsout for CSS
r? @hallvors for JS

### CODE REVIEW

**Comment by miketaylr** [comment]
Ah, need to add fixture data for this:

```
IOError: [Errno 2] No such file or directory: u'/home/travis/build/webcompat/webcompat.com/tests/fixtures/api/issues/396.json'
```

**Comment by miketaylr** [comment]
gif (ignore the compression artifacts)

nsfw

**Comment by magsout** [comment]
My first review for the mozLondon week ? 👍

**Comment by miketaylr** [comment]
> My first review for the mozLondon week ? 👍

Heh yeah, no rush!

File: webcompat/static/js/lib/issues.js [hunk 1]
```diff
@@ -395,7 +405,12 @@ issues.MainView = Backbone.View.extend({
   fetchModels: function() {
     var headersBag = {headers: {'Accept': 'application/json'}};
     this.issue.fetch(headersBag).success(_.bind(function() {
-      _.each([this.title, this.metadata, this.body, this.labels,
+      // _.find() will return the object if found (which is truthy),
+      // or undefined if not found (which is falsey)
+      this._isNSFW = _.find(this.issue.get('labels'),
```
Comment (hallvors) [inline webcompat/static/js/lib/issues.js]: Might be a good idea to add ``!!`` to turn truthy and falsy values into actual booleans.

File: tests/functional/issues-non-auth.js [hunk 2]
```diff
@@ -151,6 +151,40 @@ define([
...
+    },
+
+    'NSFW images are blurred': function() {
+      return this.remote
+        .setFindTimeout(intern.config.wc.pageLoadTimeout)
+        .get(require.toUrl(url('/issues/396')))
+        .sleep(1000)
+        .findByCssSelector('.js-Issue-commentList .js-Comment-content p').getAttribute('class')
+        .then(function(className) {
+          assert.include(className, 'nsfw');
```
Comment (magsout) [inline tests/functional/issues-non-auth.js]: I thinkg `wc-Comment-content-nsfx` would better for ClassName .. ?

File: webcompat/static/css/development/components/markdown.css [hunk 3]
```diff
@@ -39,6 +41,35 @@
     max-height: 100%;
   }
 
+  .wc-Markdown .nsfw {
+    position: relative;
+  }
+
+  .wc-Markdown .nsfw::after {
```
Comment (magsout) [inline webcompat/static/css/development/components/markdown.css]: I think add `cursoir: pointer`  would be cool..

**Comment by magsout** [comment]
Except for `className`, lgtm, nice work @miketaylr

File: tests/functional/issues-non-auth.js [hunk 4]
```diff
@@ -151,6 +151,40 @@ define([
...
+        .end()
+        .findByCssSelector('.js-Issue-commentList .js-Comment-content p').getAttribute('class')
+        .then(function(className) {
+          assert.include(className, 'nsfw--display');
+        }).click()
+        .end()
+        .findByCssSelector('.js-Issue-commentList .js-Comment-content p').getAttribute('class')
+        .then(function(className) {
+          assert.notInclude(className, 'nsfw--display');
+        })
```
Comment (hallvors) [inline tests/functional/issues-non-auth.js]: Maybe add ``assert.include(className, 'nsfw');`` too

**Comment by hallvors** [comment]
Indeed, :up:

**Comment by miketaylr** [comment]
Cool, thx for review @hallvors and @magsout. Given a few minutes I'll push the follow up commits here.

File: webcompat/static/js/lib/issues.js (hunk 1 above)
Comment (miketaylr) [inline webcompat/static/js/lib/issues.js]: Done.

File: webcompat/static/css/development/components/markdown.css (hunk 3 above)
Comment (miketaylr) [inline webcompat/static/css/development/components/markdown.css]: Ah yeah. Good point.

File: tests/functional/issues-non-auth.js (hunk 2 above)
Comment (miketaylr) [inline tests/functional/issues-non-auth.js]: Done.

**Comment by e1218520** [comment]
thx

**Comment by hallvors** [comment]
Travis seems unhappy about new test(s)?

FAIL: main - Issues - Clicking NSFW images toggles between blurry and not-blurry (1488ms)
AssertionError: expected '.wc-Comment-content-nsfw' to include 'wc-Comment-content-nsfw--display'

**Comment by miketaylr** [comment]
> Travis seems unhappy about new test(s)?

Yeah, hmm. Let me re-test locally and see what's up.

**Comment by miketaylr** [comment]
Bah, finally fixed the error of the failing test -- I had a stray `.` in `addClass`, so, yay, failing tests revealed a bug. All green now, so let's merge.

**Pull request merged**